homeassistant/components/rainmachine/binary_sensor.py

```diff
@@ -158,17 +158,17 @@ class CurrentRestrictionsBinarySensor(RainMachineEntity, BinarySensorEntity):
     def update_from_latest_data(self) -> None:
         """Update the state."""
         if self.entity_description.key == TYPE_FREEZE:
-            self._attr_is_on = self.coordinator.data["freeze"]
+            self._attr_is_on = self.coordinator.data.get("freeze")
```

Comment:
Is it possible that you don't even offer these entities if the data is not provided ?

Reply:
Definitely. The only challenges that I would now have to go back and remove orphaned entities for people with first generation controllers. Is adding that to this PR too much for beta?

Reply:
Yeah, let's do that for next release.

Reply:
If a device says unavailable because it's orphaned or unknown because the attr_is_on is None, that's not too much different annoying :)

Reply:
Adding to my todo list for next release. Thank you!